crates/prune/src/pruner.rs

```diff
@@ -548,41 +562,23 @@ impl<DB: Database> Pruner<DB> {
         // This ensures that in future pruner runs we can prune all these receipts between the
         // previous `lowest_block_with_distance` and the new one using
         // `get_next_tx_num_range_from_checkpoint`.
+        //
+        // Only applies if we were able to prune everything intended for this run, otherwise the
+        // checkpoing is the `last_pruned_block`.
         let prune_mode_block = self
             .modes
             .contract_logs_filter
-            .lowest_block_with_distance(tip_block_number, pruned_block)?
+            .lowest_block_with_distance(tip_block_number, initial_last_pruned_block)?
```

Comment:
is it correct? shouldn't it be `last_pruned_block`?

Reply:
this should be correct, this argument works as a lower bound.

if we set it to `last_pruned_block` something of the kind would happen:

example:
* `(distance(200), addrA)` 
* contract log pruner goes until distance(128).
* `last_pruned_block` would be the block from distance(128)

Which would mean that we'd never go through the blocks between  `distance(200) -> distance(128)`  for the `tip` of this particular run.